### PR TITLE
Niloofar Sadeghi / Improve live chat performance

### PR DESCRIPTION
**Pull request by niloo-fs**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [x] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [x] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/hooks/use-livechat.js [hunk 1]
```diff
@@ -1,14 +1,11 @@
 import React from 'react'
 import { getClientInformation, getDomain, getUTMData, isBrowser } from 'common/utility'
 
-export const useLivechat = () => {
+export const useLivechat = (firstLoadOpen) => {
```
Comment (yashim-deriv) [inline src/components/hooks/use-livechat.js:4]:
```suggestion
export const useLivechat = (load_livechat) => {
```
`snake_case` for arguments. recommendation for a clearer name

File: src/components/layout/livechat.js [hunk 2]
```diff
@@ -46,32 +48,57 @@ const StyledLiveChat = styled.div`
...
+    const url_params = new URLSearchParams((isBrowser() && window.location.search) || '')
+    const is_livechat_query = url_params.get('is_livechat_open')
+    const [is_loading, setIsLoading] = useState(false)
     const [is_livechat_hover, setLivechatHover] = useState(false)
-    const [is_livechat_interactive, LC_API] = useLivechat()
+    const [firstLoadOpen, setFirstLoadOpen] = useState(false)
+    const [is_livechat_interactive, LC_API] = useLivechat(firstLoadOpen)
     const { is_eu_country } = React.useContext(DerivStore)
 
+    React.useEffect(() => {
```
Comment (yashim-deriv) [inline src/components/layout/livechat.js:59]:
```suggestion
    useEffect(() => {
```

File: src/components/layout/livechat.js [hunk 3]
```diff
@@ -46,32 +48,57 @@ const StyledLiveChat = styled.div`
...
+        if (is_livechat_interactive) {
+            setIsLoading(false)
+        }
+    }, [is_livechat_interactive])
+
+    React.useEffect(() => {
```
Comment (yashim-deriv) [inline src/components/layout/livechat.js:65]:
```suggestion
    useEffect(() => {
```

File: src/components/layout/livechat.js [hunk 4]
```diff
@@ -46,32 +48,57 @@ const StyledLiveChat = styled.div`
...
 
 const LiveChat = ({ is_banner_shown }) => {
+    const url_params = new URLSearchParams((isBrowser() && window.location.search) || '')
+    const is_livechat_query = url_params.get('is_livechat_open')
+    const [is_loading, setIsLoading] = useState(false)
     const [is_livechat_hover, setLivechatHover] = useState(false)
-    const [is_livechat_interactive, LC_API] = useLivechat()
+    const [firstLoadOpen, setFirstLoadOpen] = useState(false)
+    const [is_livechat_interactive, LC_API] = useLivechat(firstLoadOpen)
     const { is_eu_country } = React.useContext(DerivStore)
```
Comment (yashim-deriv) [inline src/components/layout/livechat.js:57]:
```suggestion
    const { is_eu_country } = useContext(DerivStore)
```

**Review by yashim-deriv — COMMENTED** [review]
Everything looks good. Just fix the `React.` imports to follow deriv-com's code guidelines

File: src/pages/livechat/index.js [hunk 5]
```diff
@@ -24,23 +24,15 @@ const CoverMinimizeButton = styled.div`
 `
 
 const LiveChatPage = () => {
-    const [is_livechat_interactive, LC_API] = useLivechat()
+    const [firstLoadOpen, setFirstLoadOpen] = useState(false)
```
Comment (sean-binary) [inline src/pages/livechat/index.js:27]:
```suggestion
    const [first_load_open, setFirstLoadOpen] = useState(false)
```

**Review by yashim-deriv — APPROVED** [review]
LGTM

**Pull request merged**